langchain4j/src/main/java/dev/langchain4j/data/document/splitter/HierarchicalDocumentSplitter.java

```diff
@@ -28,11 +28,13 @@ public abstract class HierarchicalDocumentSplitter implements DocumentSplitter {
             new DocumentBySentenceSplitter(1, 0, null, null);
 
     private static final String INDEX = "index";
+    private static final String START_INDEX = "startIndex";
```

Comment:
```suggestion
    private static final String START_INDEX = "start_index";
```
To keep it in sync with the rest of metadata fields (e.g. "file_name", "absolute_directory_path", etc).

I am not sure about `start_index` name though, it might be not clear what is the difference/relationshop between `start_index` and `index`. Did you consider other options?

Reply:
The reason for start_index was to mimic the python library. Happy to change it to whatever you prefer however.

Reply:
What about `char_index_start` or `char_start_index` or some combination of these 3 words?